### PR TITLE
extract route enhancer into trait

### PR DESCRIPTION
**Pull request by dbu**
continuation of #155

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 Changelog
 =========
 
+2.0.0 (unreleased)
+------------------
```
Comment (wouterj) [inline CHANGELOG.md]: please remove the header from the CHANGELOG file in dev. Adding the header is part of the release checklist.

**Comment by dbu** [comment]
rebased to get your fix in, and fixed the changelog

**Comment by timplunkett** [comment]
Looks all but identical. I swapped out my temporary version of this and ran tests, 100% passes.
Thanks!

**Comment by dbu** [comment]
excellent! we will release 2.0 stable in the next days.

**Pull request merged**